ctl/src/command.rs

```diff
@@ -165,31 +167,33 @@ pub fn dump_state(mut channel: Channel<CommandRequest,CommandResponse>, timeout:
           },
           CommandStatus::Error => {
             if json {
-              print_json_response(&message.message);
+              print_json_response(&message.message)?;
+              return Ok(());
```

Comment:
An error should be returned here, the dump_state didn't work

Reply:
The `return Ok(());` messes up the returned value. I corrected it with

```rust
if json {
  print_json_response(&message.message)?;
} 
bail!("could not dump proxy state: {}", message.message);
```

To return an Err() anyway.